elyra/pipeline/airflow/component_parser_airflow.py

```diff
@@ -46,149 +40,368 @@ class AirflowComponentParser(ComponentParser):
     component_platform: RuntimeProcessorType = RuntimeProcessorType.APACHE_AIRFLOW
 
     def parse(self, registry_entry: SimpleNamespace) -> Optional[List[Component]]:
-        components: List[Component] = list()
+        components: List[Component] = []
 
         component_definition = registry_entry.component_definition
         if not component_definition:
             return None
 
-        # Parse the component definition for all defined classes
-        component_classes = self._get_all_classes(component_definition)
+        # Parse the component definition for all defined Operator classes
+        try:
+            parsed_class_nodes = self._parse_all_classes(component_definition)
+            num_operator_classes = len(parsed_class_nodes)
+        except Exception as e:
+            self.log.error(f"Content associated with identifier '{registry_entry.component_identifier}' "
+                           f"could not be parsed: {e}. Skipping...")
+            return None
 
-        for component_class, component_content in component_classes.items():
-            # Create a Component object for each class
-            component_properties = self._parse_properties(component_content)
+        for component_class, content in parsed_class_nodes.items():
+            if not content.get('init_function'):
+                # Without the init function, class can't be parsed for properties
+                # TODO is this the correct functionality? see classes that derive from PythonOperator
```

Comment:
This comment can probably be removed. This case will be covered in a follow-up PR as noted in #2414